### DISCUSSION TITLE
Getting Node info from swmmio

### DISCUSSION THREAD

**Discussion by bowguy** [pyswmm/swmmio#237] [General]
I am struggling to get node model results from swmmio functions.  Starting with the easiest I ran the example from the Latte_art_double_shot.
Everything runs as expected and I get the resultant test.html. When you click on a link, you get detailed model results including:
Type: CONDUIT
MaxQ: 18.3
MaxDay: 0
MaxHr: 04:02
MaxV: 9.87
MaxQPerc: 0.62
MaxDPerc: 0.57
Name: 16

However, clicking on a node only gives limited results:
19
InvertElev: 1010
MaxDepth: 3
InitDepth: 0
SurchargeDepth: 0
PondedArea: 0
Name: 19

None of the simulation results appear in the file.  Running EPA SWMM, node 19:

  ******************
  Node Depth Summary
  ******************
  
  ---------------------------------------------------------------------------------
                                 Average  Maximum  Maximum  Time of Max    Reported
                                   Depth    Depth      HGL   Occurrence   Max Depth
  Node                 Type         Feet     Feet     Feet  days hr:min        Feet
  ---------------------------------------------------------------------------------
  9                    JUNCTION     0.09     0.57  1000.57     0  04:01        0.57
  10                   JUNCTION     0.32     3.00   998.00     0  02:17        3.00
  13                   JUNCTION     0.06     0.41   995.41     0  04:01        0.41
  14                   JUNCTION     0.06     0.46   990.46     0  04:01        0.46
  15                   JUNCTION     0.15     1.15   988.15     0  04:01        1.15
  16                   JUNCTION     0.17     1.15   986.15     0  04:01        1.14
  17                   JUNCTION     0.18     1.14   981.14     0  04:02        1.14
  19                   JUNCTION     0.03     0.22  1010.22     0  04:01        0.22
  20                   JUNCTION     0.03     0.22  1005.22     0  04:02        0.22
  21                   JUNCTION     1.16     2.00   992.00     0  02:55        2.00
  22                   JUNCTION     1.11     1.58   988.58     0  04:02        1.58
  23                   JUNCTION     0.05     0.35   990.35     0  04:01        0.35
  24                   JUNCTION     0.18     1.14   985.14     0  04:01        1.14
  18                   OUTFALL      0.17     1.07   976.07     0  04:02        1.06

How can we get the node simulation values from the swmmio.create_map function?  I couldn't find any options available for it.
Thanks

**Comment by bemcdonnell** [comment]
I think you’re looking for node.statistics and  conduit.statistics

http://pyswmm.github.io/pyswmm/reference/api/api/pyswmm.nodes.Node.statistics.html#pyswmm.nodes.Node.statistics

Call this after the simulation loop but before exiting the “with” context manager

**Comment by bowguy** [comment]
node.statistics is great but it seems to be tricky to also get the node id and coordinates.  I need to convert the info to geojson which is what swmmio.create_map function does except as noted above does not include the node simulation results.
I am creating the geojson files i need via:

simulation_info = swmmio.Model(sim_file, crs='epsg:2232')
simulation_info.to_crs("EPSG:4326")

conduits = swmmio.elements.Links(
    simulation_info,
    inp_sections=['conduits'],
    rpt_sections=['Link Flow Summary']
    ,columns=['MaxD','MaxHr','MaxDPerc','MaxQ', 'MaxQPerc','MaxV','InletNode','OutletNode','Type', 'coords']
).geojson

nodes = swmmio.elements.Nodes(
    simulation_info,
    inp_sections=['junctions', 'storage', 'outfalls'],
    rpt_sections=['Node Depth Summary', 'Node Inflow Summary']
    ,columns=[ 'MaxDay', 'MaxHr', 'MaxHGL', 'SurchargeDepth', 'MaxTotalInflow', 'MaxLatInflow', 'coords']
).geojson_

What would be nice is if I could specify that geojson data in the swmmio.create_map function.  I am trying to get something like:

https://mhfd-ns5data.trilynx-novastar.systems/mhfd/data/json/MWPS/Niver_105592_v02_design_100yr.html
 in a simple call

**Reply by bemcdonnell** [reply]
@bowguy, sorry I should have read it a little closer.  @aerispaha is the human to answer this question 🫣

**Reply by bowguy** [reply]
I am testing a pull request now to pass custom geojson files to create_map

**Reply by bowguy** [reply]
Pull request submitted .  Looks good here.  See above link directly created by create_map
